### PR TITLE
Private cache improvements

### PR DESCRIPTION
**Pull request by nfuller**
See commit comment. Minor improvements to the cache strategy now that OkHttp is a private, not a shared cache.

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/CacheControl.java [hunk 1]
```diff
@@ -257,6 +264,10 @@ private String headerValue() {
     return result.toString();
   }
 
+  public boolean isPrivate() {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/CacheControl.java]: nit: define this with the other accessors

**Comment by swankjesse** [comment]
LGTM

File: okhttp/src/main/java/com/squareup/okhttp/internal/http/CacheStrategy.java [hunk 2]
```diff
@@ -67,10 +67,13 @@ public static boolean isCacheable(Response response, Request request) {
       case HTTP_MOVED_TEMP:
       case HTTP_TEMP_REDIRECT:
         // These codes can only be cached with the right response headers.
+        // http://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html#sec13.4
+        // s-maxage is not checked because OkHttp is a private cache that should ignore s-maxage as
+        // per http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9.3
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/internal/http/CacheStrategy.java]: There's new RFCs! The new hotness for caching is here:

http://tools.ietf.org/html/rfc7234
(and friends)

**Comment by nfuller** [comment]
Updated reference. The new RFC doesn't spell it out as neatly. I haven't read the newer RFC in any detail so please do confirm this conforms.

**Pull request merged**